packages/frontend/src/state/lp/hooks.ts

```diff
@@ -19,6 +19,78 @@ const POOL_FEE = 3000
 const MAX_INT_128 = new BigNumber(2).pow(128).minus(1).toFixed(0)
 const x96 = new BigNumber(2).pow(96)
 
+/*** ACTIONS ***/
+
+// Opening a mint and LP position and depositing
+export const useOpenPositionDeposit = () => {
+  const { squeethPool } = useAtomValue(addressesAtom)
+  const address = useAtomValue(addressAtom)
+  const contract = useAtomValue(controllerHelperHelperContractAtom)
+  const handleTransaction = useHandleTransaction()
+  const getDebtAmount = useGetDebtAmount()
+  const squeethPoolContract = useAtomValue(squeethPoolContractAtom)
+  const isWethToken0 = useAtomValue(isWethToken0Atom)
+  const openPositionDeposit = useAppCallback(
+    async (squeethToMint: BigNumber, lowerTickInput: number, upperTickInput: number, vaultId: number, collatRatio: number, slippage: number, withdrawAmount: number, onTxConfirmed?: () => void) => {
+      if (!contract || !address || !squeethPoolContract) return null
+      
+      const mintWSqueethAmount = fromTokenAmount(squeethToMint, OSQUEETH_DECIMALS)
+      const ethDebtPromise = getDebtAmount(mintWSqueethAmount)
+      const poolStatePromise = getPoolState(squeethPoolContract)
+
+      // Calculate prices from ticks
+      const [ethDebt, { tick, tickSpacing }] = await Promise.all([ethDebtPromise, poolStatePromise])
+      const lowerTick = nearestUsableTick(lowerTickInput, Number(tickSpacing))
+      const upperTick = nearestUsableTick(upperTickInput, Number(tickSpacing))
+      const sqrtLowerPrice = new BigNumber(TickMath.getSqrtRatioAtTick(lowerTick).toString()).div(x96)
+      const sqrtUpperPrice = new BigNumber(TickMath.getSqrtRatioAtTick(upperTick).toString()).div(x96)
+      const squeethPrice = isWethToken0 ? new BigNumber(1).div(new BigNumber(TickMath.getSqrtRatioAtTick(Number(tick)).toString()).div(x96).pow(2))
+                                            : new BigNumber(TickMath.getSqrtRatioAtTick(Number(tick)).toString()).div(x96).pow(2)
+      const sqrtSqueethPrice = squeethPrice.sqrt()
+      const collateralToMint = ethDebt.multipliedBy(collatRatio)
+
+      // Lx = x * (sqrtSqueethPrice * sqrtUpperPrice) / (sqrtUpperPrice - sqrtSqueethPrice)
+      // y = Lx * (sqrtSqueethPrice - sqrtLowerPrice)
+      const liquidity = mintWSqueethAmount.times(sqrtSqueethPrice.times(sqrtUpperPrice)).div(sqrtUpperPrice.minus(sqrtSqueethPrice))
```

Comment:
I think we need a (p < pa) case for liquidity (we can exclude the case where oSQTH amount is zero). 

x = oSQTH amount
y = ETH amount

For liquidity L:

p < pa: L = x * (sqrtLowerPrice * sqrtUpperPrice) / (sqrtUpperPrice - sqrtLowerPrice)
pa < p < pb: L = x * (sqrtSqueethPrice * sqrtUpperPrice) / (sqrtUpperPrice - sqrtSqueethPrice)
(ignoring p>pb because then squeeth amount is zero)

Then collateralToLp below covers all the cases

Reply:
Maybe it's easiest to have all three cases for liquidity and collateralToLp and throw an error if p>pb

Reply:
@alpinechicken Why can we exclude the case where oSQTH amount is 0? What if someone wants to open an all ETH position?

Reply:
maybe just because it would be a bit odd to just LP eth, but I guess there would be reasons. all three makes sense

Reply:
for p > pb:
L = y/(sqrt(pb)-sqrt(pa)

Reply:
depending on isWethToken0, isn't it possible that 100% oSQTH is when we are below lowerTick instead of above upperTick?

Reply:
And yeah don't we have to calculate liquidity differently depending on what the oSQTH price is?

Reply:
(can get a negative liquidity for example if squeeth price is > upper tick price)